File: projects/10/ruby/Duck.rb
# Duck.rb | Defines the Duck class which inherits attributes and methods
#   from the Bird superclass.
#
# Begun by: Dr. Adams, for CS 214 at Calvin College.
# Completed by: Quentin Barnes
# Date: May 07, 2019
####################################################

require './FlyingBird.rb'

class Duck < FlyingBird

    def call
        'Quack!'
    end

end

File: projects/10/java/Duck.java
/* Duck.java provides a Duck subclass of Bird.
 *
 * Begun by: Prof. Adams, for CS 214 at Calvin College.
 * Completed by: Quentin Barnes
 * Date: May 7, 2019
 ******************************************************/

public class Duck extends FlyingBird {

  /*
   * explicit constructor Receive: name, a String PostCond: myName == name.
   */

  public Duck() {
    super();
  }

  public Duck(String name) {
    super(name);
  }

  /*
   * A Duck's Call Return: a duck-call ("Quack!").
   */

  public String call() {
    return "Quack!";
  }

}
